### ISSUE TITLE
isValidSetup might double-count retiring creeps

### ISSUE THREAD

**Issue by karlthepagan** [ScreepsOCS/screeps.behaviour-action-pattern#943]
When using measureByHome creeps relating to that room are only counted if they are not within retirement age. There is no corresponding adjustment for globally measured creeps.

When using maxPerFlag there is an "extra body" counted in the number of creeps allowed to spawn as a replacement for the retiring creep.

When these techniques are used together an extra creep could be spawned beyond the replacement count.

**Comment by karlthepagan** [comment]
I'm 99% sure this is correct.

It appears to affect sourceKillers.

**Comment by karlthepagan** [comment]
I do not think that its appropriate for any maxPerFlag to measure by home regardless of this issue.

**Comment by karlthepagan** [comment]
After disabling measureByHome, a few hours later
https://screeps.com/a/#!/room/E75S6 is still getting 2 SKK's for one flag, not 4+ like before but better.

**Comment by karlthepagan** [comment]
The TTL's are 1120 vs 1093 so I don't think they were spawned in the same tick.

**Comment by karlthepagan** [comment]
This time TTL's are 797 and 801.

**Comment by chobobobo** [comment]
What is their spawn and spawn room?

**Comment by karlthepagan** [comment]
E74S7 and E76S7 are the two spawn rooms.

**Comment by karlthepagan** [comment]
1365, 1325 - still 40 ticks apart but two spawns.
